.repo/sample-list-generator/src/api-detector.ts

```diff
@@ -0,0 +1,147 @@
+import { IApiItem, TApiTypeResult, TExtensionApiMap } from './types';
+import * as babel from '@babel/core';
+import fs from 'fs/promises';
+import { getAllFiles, singularize } from './utils';
+
+let EXTENSION_API_MAP: TExtensionApiMap = {};
+(async () => {
+  try {
+    let _EXTENSION_API_MAP = await fs.readFile('./extension-apis.json', 'utf8');
+    EXTENSION_API_MAP = JSON.parse(_EXTENSION_API_MAP);
+  } catch (error) {}
```

Comment:
What's the thinking behind the try/catch? Throwing an error and stopping the script is probably better.